### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by matboivin**
GM == group messages  
DM == direct messages  
channel == both

Additions:
- GM and DM are loaded on mount according to the channels the user joined
- The user can send a new message (but need to check message full of whitespaces only :warning:)
- The last message preview is updated
- You can create a GM
- The DM creation is for now the same as a GM, it's wip
- Password-protected channels: the User's input is checked

TODO:
- Create a new DM by searching in friends
- Add a User to a group
- Channel roles: owner and admins
- Discard input message if full of whitespaces only
- Add every users to public channel upon creation
- Sockets
- For now I assume the User joined channels so I have a few more fixes to do
- and a lot more

### CODE REVIEW

File: src/nextjs-front/pages/users/[id].tsx [hunk 1]
```diff
@@ -167,36 +166,14 @@ const UserProfilePage: NextPageWithLayout = ({}) => {
   );
 
   const router = useRouter();
-  
+
+  /* Send DM to user */
   const handleMessage = () => {
-    setChatView('dm', 'direct message', { targetUsername: userId });
+    setChatView('dm', 'direct message', { targetUsername: userData.username });
```
Comment (busshi) [inline src/nextjs-front/pages/users/[id].tsx:172]: What happen if the usern changes his username ? Using the id to refer is the only data that does not change ? Why use the username ?

File: src/nextjs-front/pages/users/[id].tsx [hunk 2]
```diff
@@ -221,6 +198,31 @@ const UserProfilePage: NextPageWithLayout = ({}) => {
     });
   }
 
+  /* Send friendship invite */
+  const requestFriend = async (id: string, username: string) => {
+    const reqSent = await fetch (`/api/users/${getUserData().id}`, {
```
Comment (busshi) [inline src/nextjs-front/pages/users/[id].tsx:203]: You have to add the requestSent too to the sender. This field is used to check for friends page. It is better to add this here too not to "break" something I think.

File: src/nestjs-back/src/auth/auth.service.ts [hunk 3]
```diff
@@ -97,4 +95,13 @@ export class AuthService {
       access_token: this.jwtService.sign({ sub: ''+duoQuadraUser.id })
     });
   }
+
+  async getUserFromAuthToken(token: string) {
```
Comment (busshi) [inline src/nestjs-back/src/auth/auth.service.ts:99]: Do you have to check jwt here ? It is already check at the signin step.

**Review by busshi — APPROVED** [review]
Good work !!!

File: src/nestjs-back/src/auth/auth.service.ts (hunk 3 above)
Comment (matboivin) [reply src/nestjs-back/src/auth/auth.service.ts:99]: I think I will need your help on this one and auth-related code. We should catch up later.

File: src/nextjs-front/pages/users/[id].tsx (hunk 1 above)
Comment (matboivin) [reply src/nextjs-front/pages/users/[id].tsx:172]: This is the name displayed on the DM header (next to the status icon):

Screenshot at 2022-03-23 10-24-09

**Pull request merged**